### ISSUE TITLE
encrypt with master-password

### ISSUE THREAD

**Issue by rigid** [tobykurien/WebApps#21]
an awesome feature would be password encrypted webapp-data (cache, cookies, etc.) so that the app cannot be used, without entering a password.

**Comment by untitaker** [comment]
I feel like this is the responsibility of Android's full-disk encryption.

**Comment by rigid** [comment]
you're right. I just thougt about disabling other people from opening the app (which would make little sense if you could copy all raw data via adb for example).
This might aswell be the responsibility of the lockscreen. t'was just a random thought, feel free to close the issue, if you like.

**Comment by tobykurien** [comment]
I think it's potentially a useful feature - you're logged into a bunch of sensitive webapps (e.e.g social media, etc.) and then want to hand your phone over to someone so they can look at photos, then you want to know that they can't easily launch WebApps and post/read stuff.

**Comment by rigid** [comment]
:+1:

**Comment by tobykurien** [comment]
The app can be protected with a lockscreen: https://stackoverflow.com/questions/25562899/service-for-lock-screen-before-opening-app#35338758

**Comment by trymeouteh** [comment]
The ability to have a specific web app to require a pin or password will be handy. Some apps you may want extra security measures on.
